core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -105,15 +105,29 @@ private[ui] class StagePage(parent: StagesTab, store: AppStatusStore) extends We
     val stageAttemptId = parameterAttempt.toInt
 
     val stageHeader = s"Details for Stage $stageId (Attempt $stageAttemptId)"
-    val stageData = parent.store
-      .asOption(parent.store.stageAttempt(stageId, stageAttemptId, details = false))
-      .getOrElse {
+    var stageDataTuple: Tuple2[StageData, Seq[Int]] = null
+    try {
```

Comment:
I think you could use options here instead of try and null checks to make this cleaner

Reply:
Done, thanks!